core/src/main/java/com/linecorp/armeria/client/DnsResolverGroupBuilder.java

```diff
@@ -361,13 +376,17 @@ RefreshingAddressResolverGroup build(EventLoopGroup eventLoopGroup) {
                     new DefaultDnsQueryLifecycleObserverFactory(
```

Comment:
`observerFactory` is not used when `disableDnsQueryLifecycleObserverFactory` is false so we should remove it. 😄